src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/OutputProducer.cs

```diff
@@ -112,8 +113,12 @@ private Task FlushAsyncAwaited(WritableBufferAwaitable awaitable)
                     });
                 }
             }
+            await _flushTcs.Task;
 
-            return _flushTcs.Task;
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
If you run WriteAsync without CT, and on with CT I didn't want to cancel first task